packages/camera/camera_android_camerax/lib/src/java_object.dart

```diff
@@ -26,19 +27,31 @@ class JavaObject {
         );
 
   /// Global instance of [InstanceManager].
-  static final InstanceManager globalInstanceManager = InstanceManager(
-    onWeakReferenceRemoved: (int identifier) {
-      JavaObjectHostApiImpl().dispose(identifier);
-    },
-  );
+  static final InstanceManager globalInstanceManager = _initInstanceManager();
 
-  /// Pigeon Host Api implementation for [JavaObject].
-  final JavaObjectHostApiImpl _api;
+  static InstanceManager _initInstanceManager() {
+    WidgetsFlutterBinding.ensureInitialized();
+    // Clears the native `InstanceManager` on initial use of the Dart one.
+    InstanceManagerHostApi().clear();
+    return InstanceManager(
+      onWeakReferenceRemoved: (int identifier) {
+        JavaObjectHostApiImpl().dispose(identifier);
+      },
+    );
+  }
 
-  /// Release the reference to a native Java instance.
-  static void dispose(JavaObject instance) {
-    instance._api.instanceManager.removeWeakReference(instance);
+  /// Release the weak reference to the [instance].
+  ///
+  /// Uses the [globalInstanceManager] by default.
+  static void dispose(
+    JavaObject instance, {
+    @visibleForTesting InstanceManager? instanceManager,
+  }) {
+    (instanceManager ?? globalInstanceManager).removeWeakReference(instance);
```

Comment:
I still need to fix this in `wrapper_example`. I think this method definition is better as:

```dart
/// Release the reference to a native Java instance.
static void dispose(JavaObject instance) {
  instance._api.instanceManager.removeWeakReference(instance);
}
```